plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/model/LibraryFile.java

```diff
@@ -104,4 +112,21 @@ public int hashCode() {
     return this.mavenCoordinates.toStringCoordinates().hashCode();
   }
 
+  /**
+   * Check Maven Central to find the latest release version of this artifact.
+   * This check is made at most once. Subsequent checks are no-ops.
+   */
+  void updateVersion() {
+    if (!fixedVersion) {
+      // todo need method to get latest nonrelease version instead for alphas and betas
+      ArtifactVersion remoteVersion = ArtifactRetriever.DEFAULT.getLatestArtifactVersion(
+          mavenCoordinates.getGroupId(), mavenCoordinates.getArtifactId());
+      if (remoteVersion != null) {
+        String updatedVersion = remoteVersion.toString(); 
+        mavenCoordinates = mavenCoordinates.toBuilder().setVersion(updatedVersion).build();
+      }
+      fixedVersion = true;
```

Comment:
Mark it fixed even if `remoteVersion` is `null`?

Reply:
Yes. The goal is to check once and only once on first load. After that we don't want the version to change. This is important for serialization and reproducibility.